packages/whenable/src/module.js

```diff
@@ -0,0 +1,27 @@
+/* global globalThis */
+// @ts-check
+import { makeWhen } from './when.js';
+import { prepareWhenableKits } from './whenable.js';
+import { prepareWatch } from './watch.js';
+import makeE from './E.js';
+
+/**
+ * @param {import('@agoric/base-zone').Zone} zone
+ * @param {object} [powers]
+ * @param {(reason: any) => boolean} [powers.rejectionMeansRetry]
+ * @param {(p: PromiseLike<any>, watcher: import('./watch.js').PromiseWatcher, ...args: unknown[]) => void} [powers.watchPromise]
+ */
+export const prepareWhenableModule = (zone, powers) => {
+  const { rejectionMeansRetry = _reason => false, watchPromise } = powers || {};
+  const { makeWhenableKit, makeWhenablePromiseKit } = prepareWhenableKits(zone);
+  const when = makeWhen(makeWhenablePromiseKit, rejectionMeansRetry);
+  const watch = prepareWatch(
+    zone,
+    makeWhenableKit,
+    watchPromise,
+    rejectionMeansRetry,
+  );
+  const E = makeE(globalThis.HandledPromise, when);
+  return harden({ E, watch, when, makeWhenableKit, makeWhenablePromiseKit });
```

Comment:
I like the way this layering of concepts is working out. But the record returned here is the normal-use public API. Good to make this as thin as possible with no loss of functionality. Could it be

```js
  return harden({ E, makeWhenableKit });
```

Since
- when and watch are moved into E
- wouldn't `E.when(whenable)` create a promise equivalent to the one that would be provided by `makeWhenablePromiseKit`?

Reply:
Great suggestion!  And you're right about composing `E.when` with `makeWhenableKit().whenable` to remove `makeWhenablePromiseKit` from the public API.  I'll still need to pass it internally to the `makeWhen` function, or else there'd be infinite recursion in the definition of `when`.

Reply:
Great! Resolving.